From 5eb36ee96aff62424988ce241701477b9cf13302 Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Thu, 26 Dec 2024 13:48:46 +0100
Subject: [PATCH 8/8] fix(NetworkTraffic): set values to minimum of 0

It seems as if TrafficStats.getXXBytes likes to output negative values when it fails
to grab values. The negative value is likely -1.
---
 .../src/com/android/systemui/statusbar/NetworkTraffic.java   | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/NetworkTraffic.java b/packages/SystemUI/src/com/android/systemui/statusbar/NetworkTraffic.java
index 3a114f3fa..bf2a4f7bf 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/NetworkTraffic.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/NetworkTraffic.java
@@ -223,7 +223,10 @@ public class NetworkTraffic extends TextView implements TunerService.Tunable,
                     rxBytes += ifaceRxBytes;
                 }
 
-                if (txBytes <= 0 && rxBytes <= 0) {
+                txBytes = Math.max(0, txBytes);
+                rxBytes = Math.max(0, rxBytes);
+
+                if (txBytes == 0 && rxBytes == 0) {
                     // Cannot capture anything through TrafficStats.
                     // This is an old kernel, not supporting eBPF.
                     // Falling back to kernel file reading.
-- 
2.34.1

